pkg/certsigner/signer.go

```diff
@@ -288,7 +336,48 @@ func (s *CertServer) HandlePostCSR(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	signedCSR, err := s.signer.Sign(csr)
+	x509CSR, err := csrutil.ParseCSR(csr)
```

Comment:
why move this out of `Sign` ?

Reply:
Because I need the profile on line 364 which is parsed by the CSR. By moving it to the handler HandlePostCSR I can parse profile per request and then use it as a condition to populate SignerCA. I need this before Sign.

Reply:
Ah, can we split that out into separate func that we can test it separately ?

Reply:
sure sounds good!

Reply:
Will add tests as well then ;)